File: src/app/shared/models/product.model.ts
import { Category } from './category.model';

export class Product {
	id: number;
	creationDate: Date | string;
	productName: string;
	categoryId: number;
	availableQuantity: number;
	price: number;
	description?: string = '';
	size: string;
	category: Category | null;

	constructor(
		id: number,
		creationDate: Date | string,
		productName: string,
		categoryId: number,
		availableQuantity: number,
		price: number,
		description: string,
		size: string,
		category: Category | null
	) {
		this.id = id;
		this.creationDate = creationDate;
		this.productName = productName;
		this.categoryId = categoryId;
		this.availableQuantity = availableQuantity;
		this.price = price;
		this.description = description;
		this.size = size;
		this.category = category;
	}
}

File: AspNet/StoreApi/StoreApi/Models/ProductModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StoreApi.Models
{
    public class ProductModel
    {
        public int Id { get; set; }

        public string ProductName { get; set; }

        public int AvailableQuantity { get; set; }

        public int Price { get; set; }

        public DateTime CreationDate { get; set; }

        public string Description { get; set; }

        public int? CategoryId { get; set; }
    }
}
